contracts/ERC721MLite.sol

```diff
@@ -0,0 +1,568 @@
+     *      startTimeUnixSeconds: 1667768000,
+     *      endTimeUnixSeconds: 1667771600,
+     *     },
+     *     {
+     *      price: 20000000000000000000,
+     *      maxStageSupply: 3000,
+     *      walletLimit: 2,
+     *      merkleRoot: 0,
+     *      startTimeUnixSeconds: 1667771600,
+     *      endTimeUnixSeconds: 1667775200,
+     *     }
+     * ]
+     */
+    function setStages(MintStageInfo[] calldata newStages) external onlyOwner {
+        uint256 originalSize = _mintStages.length;
+        for (uint256 i = 0; i < originalSize; i++) {
+            _mintStages.pop();
+        }
+
+        uint64 timestampExpirySeconds = getTimestampExpirySeconds();
+        for (uint256 i = 0; i < newStages.length; i++) {
+            if (i >= 1) {
+                if (
+                    newStages[i].startTimeUnixSeconds <
+                    newStages[i - 1].endTimeUnixSeconds + timestampExpirySeconds
+                ) {
+                    revert InsufficientStageTimeGap();
+                }
+            }
+            _assertValidStartAndEndTimestamp(
+                newStages[i].startTimeUnixSeconds,
+                newStages[i].endTimeUnixSeconds
+            );
+            _mintStages.push(
+                MintStageInfo({
+                    price: newStages[i].price,
+                    walletLimit: newStages[i].walletLimit,
+                    merkleRoot: newStages[i].merkleRoot,
+                    maxStageSupply: newStages[i].maxStageSupply,
+                    startTimeUnixSeconds: newStages[i].startTimeUnixSeconds,
+                    endTimeUnixSeconds: newStages[i].endTimeUnixSeconds
+                })
+            );
+            emit UpdateStage(
+                i,
+                newStages[i].price,
+                newStages[i].walletLimit,
+                newStages[i].merkleRoot,
+                newStages[i].maxStageSupply,
+                newStages[i].startTimeUnixSeconds,
+                newStages[i].endTimeUnixSeconds
+            );
+        }
+    }
+
+    /**
+     * @dev Gets whether mintable.
+     */
+    function getMintable() external view override returns (bool) {
+        return _mintable;
+    }
+
+    /**
+     * @dev Sets mintable.
+     */
+    function setMintable(bool mintable) external onlyOwner {
+        _mintable = mintable;
+        emit SetMintable(mintable);
+    }
+
+    /**
+     * @dev Returns number of stages.
+     */
+    function getNumberStages() external view override returns (uint256) {
+        return _mintStages.length;
+    }
+
+    /**
+     * @dev Returns maximum mintable supply.
+     */
+    function getMaxMintableSupply() external view override returns (uint256) {
+        return _maxMintableSupply;
+    }
+
+    /**
+     * @dev Sets maximum mintable supply.
+     */
+    function setMaxMintableSupply(uint256 maxMintableSupply)
+        external
+        virtual
+        onlyOwner
+    {
```

Comment:
If there's space, we may still want 

```
        if (maxMintableSupply > _maxMintableSupply) {
            revert CannotIncreaseMaxMintableSupply();
        }
```

since it helps minters be sure there can't be more supply and dilution for the collection

Reply:
I can just add it back to keep this conservative. FYI today we have `ERC721MIncreasableSupply` that allows increasing supply any way.